src/credix_lp/depository.ts

```diff
@@ -0,0 +1,301 @@
+const CREDIX_LP_INTERNAL_CREDIX_PASS_NAMESPACE = 'credix-pass';
+const CREDIX_LP_INTERNAL_PROGRAM_STATE_NAMESPACE = 'program-state';
+
+export class CredixLpDepository {
+  public constructor(
+    public readonly pda: PublicKey,
+    public readonly collateralMint: PublicKey,
+    public readonly collateralDecimals: number,
+    public readonly collateralSymbol: string,
+    public readonly depositoryCollateral: PublicKey,
+    public readonly depositoryShares: PublicKey,
+    public readonly credixProgramState: PublicKey,
+    public readonly credixGlobalMarketState: PublicKey,
+    public readonly credixSigningAuthority: PublicKey,
+    public readonly credixLiquidityCollateral: PublicKey,
+    public readonly credixSharesMint: PublicKey,
+    public readonly credixPass: PublicKey,
+    public readonly credixTreasuryCollateral: PublicKey,
+    public readonly credixMultisigKey: PublicKey,
+    public readonly credixMultisigCollateral: PublicKey,
+    public readonly credixProgramId: PublicKey
+  ) {}
+
+  public static async initialize({
+    connection,
+    uxdProgramId,
+    collateralMint,
+    collateralSymbol,
+    credixProgramId,
+  }: {
+    connection: Connection;
+    uxdProgramId: PublicKey;
+    collateralMint: PublicKey;
+    collateralSymbol: string;
+    credixProgramId: PublicKey;
+  }): Promise<CredixLpDepository> {
+    // First we need the credix global market state address
+    const credixGlobalMarketStatePromise =
+      this.findCredixGlobalMarketState(credixProgramId);
+    // Then the credix program state address
+    const credixProgramStatePromise =
+      this.findCredixProgramState(credixProgramId);
+
+    // Then schedule reading the collateral mint
+    const collateralToken = new Token(
+      connection,
+      collateralMint,
+      TOKEN_PROGRAM_ID,
+      null as unknown as Signer
+    );
+    const collateralInfoPromise = collateralToken.getMintInfo();
+
+    // Load credix IDL to be able to read onchain data
+    const provider = new AnchorProvider(
+      connection,
+      {} as Wallet,
+      AnchorProvider.defaultOptions()
+    );
+    const credixProgram = new Program<CredixIDL>(
+      credixIDL,
+      credixProgramId,
+      provider
+    );
+
+    // Now we read the results from initial finds
+    const credixGlobalMarketState = await credixGlobalMarketStatePromise;
+    const credixProgramState = await credixProgramStatePromise;
+
+    // Then we schedule reading the content of the credixGlobalMarketState onchain
+    const credixGlobalMarketStateDataPromise =
+      credixProgram.account.globalMarketState.fetchNullable(
+        credixGlobalMarketState
+      );
+    // Then we schedule reading the content of the credixProgramState onchain
+    const credixProgramStateDataPromise =
+      credixProgram.account.programState.fetchNullable(credixProgramState);
+
+    // Then we can find the depository
+    const [depository] = PublicKey.findProgramAddressSync(
```

Comment:
Question:

Why use `findProgramAddressSync` here, but `findProgramAddress` for `findCredixProgramState` for example?

Reply:
good point, fixing now 👌